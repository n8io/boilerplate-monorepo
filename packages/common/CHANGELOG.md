# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.9.0](https://github.com/n8io/boilerplate-monorepo/compare/v1.8.0...v1.9.0) (2020-12-23)

**Note:** Version bump only for package @boilerplate-monorepo/common





# [1.5.0](https://github.com/n8io/boilerplate-monorepo/compare/v1.4.0...v1.5.0) (2020-05-29)


### Features

* **feature-flags:** ‚ú® Add attribute map to feature flag requests ([24b251a](https://github.com/n8io/boilerplate-monorepo/commit/24b251ac6e10951eee9b1420cff0e106f0c707d7))





# [1.4.0](https://github.com/n8io/boilerplate-monorepo/compare/v1.3.2...v1.4.0) (2020-05-26)

**Note:** Version bump only for package @boilerplate-monorepo/common





# [1.3.0](https://github.com/n8io/boilerplate-monorepo/compare/v1.2.0...v1.3.0) (2020-05-18)


### Features

* ‚ú® Add `UserList` to users page ([fc76e9e](https://github.com/n8io/boilerplate-monorepo/commit/fc76e9e3a1dac03e512b0e8f8cae640125fd9d9a))





# [1.2.0](https://github.com/n8io/boilerplate-monorepo/compare/v1.1.1...v1.2.0) (2020-05-17)


### Features

* **ui:** ‚ú® Add user management page and respective nav links ([df3740f](https://github.com/n8io/boilerplate-monorepo/commit/df3740f0d8856467952119769fd3f18636a62fa3))





# 1.1.0 (2020-05-16)


### Bug Fixes

* ‚¨áÔ∏è Rollback versions to 1.0.0 ([e173e1a](https://github.com/n8io/boilerplate-monorepo/commit/e173e1aae513c54bb720b3dc907b95803b96e900))
* üíö Fix semantic release on builds ([08491e1](https://github.com/n8io/boilerplate-monorepo/commit/08491e17f1c8ae720e76ed03f7ea503265c9c5c7))
* üíö Update how cache keys are read ([e55eb27](https://github.com/n8io/boilerplate-monorepo/commit/e55eb2772c93778f5acd6b5bee3c4a98057a1997))
* üî• Remove extra build stuffs around version bumping ([bc77fb4](https://github.com/n8io/boilerplate-monorepo/commit/bc77fb45cba5ede97388276c38ae895ac5aaceb1))
