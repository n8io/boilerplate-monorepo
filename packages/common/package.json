{
  "name": "@boilerplate-monorepo/common",
  "version": "1.0.0",
  "private": true,
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "test": "jest",
    "test:watch": "yarn -s test --watch"
  },
  "dependencies": {
    "ramda": "^0.26.1"
  },
  "devDependencies": {
    "@babel/core": "^7.5.5",
    "@babel/preset-env": "^7.5.5",
    "babel-jest": "^24.8.0",
    "babel-plugin-macros": "^2.6.1",
    "babel-plugin-module-resolver": "^3.2.0",
    "eslint": "^6.1.0",
    "eslint-config-prettier": "^6.0.0",
    "eslint-plugin-eslint-comments": "^3.1.2",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-import-helpers": "^1.0.2",
    "eslint-plugin-jest": "^22.15.0",
    "eslint-plugin-prettier": "^3.1.0",
    "eslint-plugin-ramda": "^2.5.1"
  }
}
