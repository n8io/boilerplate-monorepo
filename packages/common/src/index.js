export { Color } from './types/color';
export { Email } from './types/email';
export { FetchPolicy } from './types/fetchPolicy';
export { LogLevel } from './types/logLevel';
export { Pagination } from './types/pagination';
export { PaginationInput } from './types/paginationInput';
export { Permission } from './types/permission';
export { Phone } from './types/phone';
export { Time } from './types/time';
export { User } from './types/user';
export { UnsafeProps } from './types/unsafeProps';
export { UserLoginInput } from './types/userLoginInput';
export { UserPasswordResetInput } from './types/userPasswordResetInput';
export { UserPasswordResetRequestInput } from './types/userPasswordResetRequestInput';
export { UserRecovery } from './types/userRecovery';
export { UserRecoveryFindInput } from './types/userRecoveryFindInput';
export { UserRecoveryNotifyInput } from './types/userRecoveryNotifyInput';
export { UserRecoveryNotifyMethod } from './types/userRecoveryNotifyMethod';
export { UserRegisterInput } from './types/userRegisterInput';
export { UserSelfSecurityUpdateInput } from './types/userSelfSecurityUpdateInput';
export { UserRole } from './types/userRole';
export { UserSelfUpdateInput } from './types/userSelfUpdateInput';
export { UserSnapshot } from './types/userSnapshot';
export {
  Validation,
  ValidationError,
  ValidationErrorKeys,
} from './types/validation';
export { Utils } from './utils';
