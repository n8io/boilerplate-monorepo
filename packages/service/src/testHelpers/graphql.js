import { PubSub } from 'apollo-server-express';
import { createTestClient } from 'apollo-server-testing';
import { subscribe as graphqlSubscribe } from 'graphql';
import { has, head, keys, path, pipe, prop, __ } from 'ramda';
import { makeServer } from '../server';
import { make as makeContext } from './context';

const makeGraphqlClient = (context) => {
  const server = makeServer({ context });
  const { query, mutate } = createTestClient(server);

  const subscribe = async (subscription, pubsub) => {
    const iterator = await graphqlSubscribe(
      server.schema,
      subscription,
      {},
      { ...context, pubsub }
    );

    if (has('errors', iterator)) return iterator;

    const { value } = await iterator.next();

    return value;
  };

  const whileSubscribed = (subscription, callback) => {
    const pubsub = new PubSub();
    const promise = subscribe(subscription, pubsub);

    callback(pubsub);
    return promise;
  };

  return {
    context,
    execMutation: mutate,
    execQuery: query,
    whileSubscribed,
  };
};

const graphqlClient = makeGraphqlClient(makeContext());

const responseToData = (response) => {
  // eslint-disable-next-line no-console
  response.errors && console.error(response.errors);
  expect(response.errors).toBeUndefined();
  expect(response.data).not.toBeNull();

  return pipe(keys, head, prop(__, response.data))(response.data);
};

const responseToErrorCode = path(['errors', 0, 'extensions', 'code']);

export {
  graphqlClient,
  makeGraphqlClient,
  responseToData,
  responseToErrorCode,
};
