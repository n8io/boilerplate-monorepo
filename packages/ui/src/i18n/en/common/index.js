export const common = {
  CAPTCHA_ERROR: 'We could not verify your humanity',
  CONFIRM_PASSWORD_MISMATCH: 'Confirm password does not match',
  DATABASE_ERROR_OCCURRED: 'An unexpected data error has occurred',
  DATE_FUTURE: 'Must be in the future',
  DATE_PAST: 'Must be in the past',
  DOES_NOT_MEET_PASSWORD_REQUIREMENTS:
    'Must contain at least one uppercase letter, lowercase letter, number, and special character',
  DOES_NOT_MEET_USERNAME_REQUIREMENTS:
    'Must only contain letters, numbers, dashes, and underscores',
  GENERIC_NETWORK_ERROR: 'No internet connection',
  INTERNAL_SERVER_ERROR: 'An unexpected error has occurred',
  INVALID_EMAIL: 'Must be a valid email address',
  RANGE_DATE: 'Must be between {{min}} and {{max}}',
  RANGE_NUMERIC: 'Must be between {{min}} and {{max}}',
  RANGE_STRING_LENGTH: 'Must be between {{min}} and {{max}} characters',
  RANGE_STRING_LENGTH_MAX: 'Must be less than or equal to {{max}} characters',
  RANGE_STRING_LENGTH_MIN: 'Must be at least {{min}} characters',
  RATE_LIMIT_EXCEEDED:
    'Too many attempts were made in a short period of time. Please wait a minute and try again.',
  REQUIRED: 'Required',
  about: 'About',
  account: 'Account',
  confirmPassword: 'Confirm Password',
  currentPassword: 'Current Password',
  dashboard: 'Dashboard',
  emailAddress: 'Email address',
  familyName: 'Last name',
  footer: `©️ {{copyrightYear}} {{name}}. All Rights reserved.`,
  givenName: 'First name',
  home: 'Home',
  loading: 'Loading',
  loadingScreenReader: 'Please wait while content is being loaded',
  login: 'Log In',
  logout: 'Log Out',
  newPassword: 'New Password',
  noDataToDisplay: 'No data to display',
  offlineDetected: `Oh noooos! It seems you've lost internet connection`,
  onlineDetected: `Sweet, you're back online`,
  password: 'Password',
  profile: 'Profile',
  proveYouAreNotARobot: "Prove You're Not a Robot",
  settings: 'Settings',
  signup: 'Sign Up',
  siteName: `{{name}} :: {{title}}`,
  username: 'Username',
};
