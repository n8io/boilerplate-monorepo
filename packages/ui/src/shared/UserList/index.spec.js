import { UserSnapshot } from '@boilerplate-monorepo/common';
import React from 'react';
import { render } from 'testHelpers';
import { UserList } from '.';

describe('<UserList/>', () => {
  const defaultProps = {
    users: [UserSnapshot.uiExample()],
  };

  const renderComponent = (overrides) =>
    render(<UserList {...defaultProps} {...overrides} />);

  test('renders properly', () => {
    const { container } = renderComponent();

    expect(container.firstChild).toMatchSnapshot();
  });
});
