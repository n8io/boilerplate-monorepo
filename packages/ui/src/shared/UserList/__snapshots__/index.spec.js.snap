// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList/> when there are users renders users properly 1`] = `
<div
  class="UserList__Container-hcXHAO eGQWk"
>
  <div
    class="UserList__Row-bzzkPz eqbTRZ"
  >
    <div
      class="UserList__Cell-heYRAP fEyKle"
    >
      USER
    </div>
    <div
      class="UserList__Cell-heYRAP fEyKle"
    >
      USERNAME
    </div>
    <div
      class="UserList__Cell-heYRAP fEyKle"
    >
      GIVEN_NAME FAMILY_NAME
    </div>
  </div>
</div>
`;
